File: SFDeped/ViewModels/MainViewModel.vb
Option Strict On

Imports System.Collections.ObjectModel
Imports MahApps.Metro.Controls

Public Class MainViewModel
    Inherits PageBase

    Private _menuItems As New ObservableCollection(Of MenuItemViewModel)
    Private _optionMenuItems As New ObservableCollection(Of MenuItemViewModel)
    Public ReadOnly Property Menu() As ObservableCollection(Of MenuItemViewModel)
        Get
            Return _menuItems
        End Get
    End Property

    Public ReadOnly Property OptionsMenu() As ObservableCollection(Of MenuItemViewModel)
        Get
            Return _optionMenuItems
        End Get
    End Property



    Public Overrides ReadOnly Property Title As String
        Get
            Return "DEPED SCHOOL FORMS - PROJECT FAST"
        End Get
    End Property

    Public Sub New()
        SetupMenu()
    End Sub

    Private Sub SetupMenu()
        _menuItems.Add(New MenuItemViewModel With {
            .Label = "Subjects",
            .NavigationPath = "Subjects",
            .Icon = New MahApps.Metro.IconPacks.PackIconFontAwesome With {
                .Kind = MahApps.Metro.IconPacks.PackIconFontAwesomeKind.BookSolid
            }
        })

        '_menuItems.Add(New MenuItemViewModel With {
        '    .Label = "Grades",
        '    .Icon = New MahApps.Metro.IconPacks.PackIconFontAwesome With {
        '        .Kind = MahApps.Metro.IconPacks.PackIconFontAwesomeKind.DigitalTachographSolid
        '    }
        '})

        _menuItems.Add(New MenuItemViewModel With {
            .Label = "Students",
            .NavigationPath = "Students",
            .NavigationType = GetType(Students),
            .Icon = New MahApps.Metro.IconPacks.PackIconFontAwesome With {
                .Kind = MahApps.Metro.IconPacks.PackIconFontAwesomeKind.PeopleArrowsSolid
            }
        })

        _optionMenuItems.Add(New MenuItemViewModel With {
            .Label = "Settings",
            .NavigationPath = "Settings",
            .NavigationType = GetType(Settings),
            .Icon = New MahApps.Metro.IconPacks.PackIconFontAwesome With {
                .Kind = MahApps.Metro.IconPacks.PackIconFontAwesomeKind.ToolboxSolid
            }
        })

    End Sub
End Class

File: SFDepedSharp/ViewModels/MainViewModel.cs
using MahApps.Metro.Controls;
using Prism.Ioc;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Controls;

namespace SFDepedSharp.ViewModels
{
    class MainViewModel : PageBase
    {
        private readonly IContainerProvider _container;

        public ObservableCollection<NavMenuItem> MenuItems { get; set; } = new();
        public ObservableCollection<NavMenuItem> OptionMenuItems { get; set; } = new();
        public override string Title => "DepEd School Forms - Project Fast";

        public MainViewModel(IContainerProvider container)
        {

            _container = container;

            SetupMenu();

        }

        private void SetupMenu()
        {
            var menuItems = _container.Resolve<ICollection<NavMenuItem>>("menuItems");
            MenuItems.AddRange(menuItems);

            var optionMenuItems = _container.Resolve<ICollection<NavMenuItem>>("optionMenuItems");
            OptionMenuItems.AddRange(optionMenuItems);
        }
    }

    public class NavMenuItem : HamburgerMenuIconItem
    {
        public Type? NavigationType { get; internal set; }
        public string? NavigationPath => NavigationType?.Name;
    }
}
